docs/dev/background-information/sql/upgrades_and_migrator_design.mdx

```diff
@@ -0,0 +1,357 @@
+presents a problem. To get around this, on minor releases we generate a
+`stitched-migration-graph.json` file. Reference links:
+[Bazel](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/dev/release/src/util.ts?L293-327),
+[Embed](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/database/migration/shared/embed.go?L15-22),
+[Generator](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/database/migration/shared/data/cmd/generator/main.go)
+
+####  Stitched Migration Graph 
+
+`stitched-migrations-graph.json` [stitches](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@v5.2.0/-/blob/internal/database/migration/stitch/stitch.go) (you can think of this as unsquashing) historic migrations using git magic, enabling the migrator to have a reference of older migrations. This serves a few purposes:
+
+1.  When jumping across multiple versions we do not have access to a
+    full record of migration definitions on migrator disk because some
+    migrations will likely have been squashed. Therefore we need a way
+    to ensure we don't miss migrations on a squash boundary. *Note we
+    can't just re-apply the root migration after a squash because some
+    schema state that\'s already represented in the root migration. This
+    means the squashed root migration isn't always idempotent.*
+
+2.  During a multiversion upgrade migrator must schedule out of band
+    migrations to be started at some version and completed before
+    upgrading past some later version. Migrator needs access to the
+    unsquashed migration definitions to know which migrations must have
+    run at the time the oob migration is triggered.
+
+In standard/`up` upgrades `stitched-migrations.json` isn't necessary. This
+is because `up` determines migrations to run by [comparing
+migrations](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/internal/database/migration/definition/definition.go?L214-244)
+listed as already run in the relevant db's `migration_logs` table directly
+to those migration definitions
+[embedded](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/blob/migrations/embed.go)
+in the migrator disk at build time for the current version, and running
+any which haven't been run. We never squash away the previous minor
+version of Sourcegraph, in this way we can guarantee the `migration_logs`
+table migrations always has migrations in common with the migration
+definitions on disk.
+
+### Database Drifts
+
+Database drift is any difference between the expected state of the
+database at a given version and the actual state of the database. How
+does it come to be? We've observed drift in customer databases for a
+number of reasons, both due to admin tampering and problems in our own
+release process:
+
+-   **A backup/restore process that didn't include all objects**: This
+    notably happened to Reddit causing their database to have no
```

Comment:
Do we want to mention a customer name here?

Reply:
Ahh good catch, I'll remove that